msprime/ancestry.py

```diff
@@ -721,113 +762,375 @@ def simulate(
+    # single-population model with a given population size.
+    if demography is None:
+        if is_dtwf:
+            # A default size of 1 isn't so smart for DTWF and almost certainly
+            # an error.
+            if population_size is None:
                 raise ValueError(
-                    "Must specify a sequence_length for simulations of a "
-                    "genome with recombination or gene conversion without "
-                    "rate map"
+                    "When using the DTWF model, the population size must be set "
+                    "explicity, either using the population_size or demography "
+                    "arguments."
                 )
+        population_size = 1 if population_size is None else float(population_size)
+        demography = demog.Demography.simple_model(population_size)
+    elif isinstance(demography, demog.Demography):
+        if population_size is not None:
+            raise ValueError("Cannot specify demography and population size")
+    else:
+        raise TypeError("demography argument must be an instance of msprime.Demography")
+    # TODO should we validate that the population sizes are positive integers
+    # for DTWF? It's not actually required, but could help catch some errors?
+    demography.validate()
 
-    sim = simulator_factory(
-        sample_size=sample_size,
-        samples=samples,
-        ploidy=ploidy,
-        length=sequence_length,
+    if initial_state is None:
+        if samples is None:
+            raise ValueError("Either samples or initial state must be specified")
+        initial_state = tskit.TableCollection(sequence_length)
+        demography.insert_populations(initial_state)
+        _parse_samples(samples, ploidy, initial_state)
+    else:
+        if samples is not None:
+            raise ValueError("Cannot specify both samples and initial_state")
+        if sequence_length != initial_state.sequence_length:
+            raise ValueError(
+                "The initial_state sequence length must agree with other parameters"
+            )
+        if len(initial_state.populations) == 0:
+            raise ValueError(
+                "initial_state tables must defined at least one population."
+            )
+
+    # For the standard code path, random_generator should always be set. But,
+    # we can also use the parser code elsewhere, so it's useful to be able
+    # to either provide a random seed, or get a good one.
+    if random_generator is None:
+        random_generator = _parse_random_seed(random_seed)
+    elif random_seed is not None:
+        raise ValueError("Cannot specify both random seed and random generator")
+
+    return Simulator(
+        tables=initial_state,
+        random_generator=random_generator,
+        recombination_map=recombination_map,
+        gene_conversion_map=gene_conversion_map,
+        gene_conversion_track_length=gene_conversion_track_length,
```

Comment:
I think it's "tract" not "track": [891 hits](https://scholar.google.com/scholar?q="gene+conversion+tract") versus [22](https://scholar.google.com/scholar?q="gene+conversion+track").

Reply:
Opened issue #1202 to track (or tract?)